### PR TITLE
fix: don't put string contents in string end token

### PR DESCRIPTION
**Pull request by brandonspark**
My previous PR at https://github.com/fwcd/tree-sitter-kotlin/pull/76 made it such that we used different logic for string scanning. I remarked that 
> Since the string internals are all hidden in the syntax tree anyways, there's no point in going to the effort of 
specifically separating the string end from string contents. If we see a bunch of quotes in a row, then we just go until
they stop appearing, then stop lexing and call it the string's end.

This is not exactly true. I was under the impression because the `tree-sitter parse` output showed just a single node for `string_literal`, that it would all be subsumed under the same thing, and it wouldn't matter exactly where you called `STRING_END` or `STRING_CONTENT`.

However, in the grammar, we see that there are indeed `string_end` and `string_start` nodes, just hidden in the parse tree.

This was causing downward consumption errors for tools like Semgrep, however, which un-hide those nodes.

So I made this change, which correctly tries to attribute string contents to only the `STRING_CONTENT`, and not `STRING_END` nodes.

So now we properly get something like:
image

in the debug parsing output for a file 
```
val sha256Digest = "SHA256"
val md5Digest = """MD5"""
```

### CODE REVIEW

**Comment by brandonspark** [comment]
@fwcd

**Comment by brandonspark** [comment]
@fwcd any plans on merging this + others? i'm a little blocked on working on kotlin otherwise

**Review by fwcd — COMMENTED** [review]
Minor comment below, otherwise this looks good, thanks for investigating!

File: src/scanner.c [hunk 1]
```diff
@@ -178,11 +193,18 @@ static bool scan_string_content(TSLexer *lexer, Stack *stack) {
...
-        advance(lexer);
-        mark_end(lexer);
-        lexer->result_symbol = STRING_END;
-        return true;
+        if (has_content) {
+          mark_end(lexer);
+          lexer->result_symbol = STRING_CONTENT;
+          return true;
+        }
+        else {
```
[the remark this answers is not in the document]
Comment (brandonspark) [reply src/scanner.c]: Thanks!

**Comment by brandonspark** [comment]
@fwcd

**Comment by fwcd** [comment]
Thanks for investigating this!

**Pull request merged**